src/pytorch_lightning/trainer/trainer.py

```diff
@@ -696,6 +696,8 @@ def fit(
 
             datamodule: An instance of :class:`~pytorch_lightning.core.datamodule.LightningDataModule`.
         """
+        if not isinstance(model, pl.LightningModule):
```

Comment:
noob q:
how is it different from typing? I know it doesn't raise any error, but we don't check for every argument.

Reply:
I can't give a good answer here. We don't have a rule written down anywhere, but we generally want to make sure we provide good informing error messages. If we didn't do a custom error message here, we would get a misleading one:

TypeError: Unwrapping the module did not yield a LightningModule, got <class 'torch.nn.parallel.distributed.DistributedDataParallel'> instead.`